scalafix-core/shared/src/main/scala/scalafix/v1/Sym.scala

```diff
@@ -0,0 +1,149 @@
+      if (sym.isLocal) Some(sym) else scala.None
+  }
+
+  object Global {
+    def unapply(sym: Sym): Option[(Sym, Sym)] =
+      if (sym.isGlobal) {
+        val owner = Sym(sym.value.owner)
+        Some(owner -> sym)
+      } else {
+        scala.None
+      }
+  }
+
+  final class Info private[scalafix] (
+      private[scalafix] val info: s.SymbolInformation
+  ) {
+    def isNone: Boolean = info.symbol.isEmpty
+    def sym: Sym = new Sym(info.symbol)
+    def owner: Sym = new Sym(info.owner)
+    def name: String = info.name
+    def kind: Kind = new Kind(info)
+    def props: Properties = new Properties(info.properties)
+    def access: Accessibility =
+      new Accessibility(info.accessibility.getOrElse(s.Accessibility()))
+
+    override def toString: String = s"Sym.Info(${info.symbol})"
+
+    // privates
+    private[scalafix] def tpe: s.Type =
+      info.tpe.getOrElse(s.Type())
+  }
+  object Info {
+    val empty = new Sym.Info(s.SymbolInformation())
+  }
+
+  final class Kind private[Sym] (info: s.SymbolInformation) {
+    def isClass: Boolean = k.isClass
+    def isObject: Boolean = k.isObject
+    def isTrait: Boolean = k.isTrait
+    def isMethod: Boolean = k.isMethod
+    def isField: Boolean = k.isField
+    def isMacro: Boolean = k.isMacro
+    def isConstructor: Boolean = k.isConstructor
+    def isType: Boolean = k.isType
+    def isParameter: Boolean = k.isParameter
+    def isTypeParameter: Boolean = k.isTypeParameter
+    def isPackage: Boolean = k.isPackage
+    def isPackageObject: Boolean = k.isPackageObject
+
+    override def toString: String = s"Sym.Kind(${info.symbol})"
+
+    // privates
+    private[this] def k = info.kind
+    private[this] def is(property: s.SymbolInformation.Property): Boolean =
+      (info.properties & property.value) != 0
+    private[this] def isSetter = info.name.endsWith("_=")
+  }
+
+  final class Properties private[Sym] (props: Int) {
+    def isAbstract: Boolean = is(p.ABSTRACT)
+    def isFinal: Boolean = is(p.FINAL)
+    def isSealed: Boolean = is(p.SEALED)
+    def isImplicit: Boolean = is(p.IMPLICIT)
+    def isLazy: Boolean = is(p.LAZY)
+    def isCase: Boolean = is(p.CASE)
+    def isCovariant: Boolean = is(p.COVARIANT)
+    def isContravariant: Boolean = is(p.CONTRAVARIANT)
+    def isStatic: Boolean = is(p.STATIC)
+    def isPrimary: Boolean = is(p.PRIMARY)
+    def isEnum: Boolean = is(p.ENUM)
+    def isVal: Boolean = is(p.VAL)
+    def isVar: Boolean = is(p.VAR)
+
+    // privates
+    private[this] def is(property: s.SymbolInformation.Property): Boolean =
+      (props & property.value) != 0
+  }
+
+  final class Accessibility private[Sym] (a: s.Accessibility) {
+    def isPrivate: Boolean = a.tag.isPrivate
+    def isPrivateThis: Boolean = a.tag.isPrivateThis
+    def isProtected: Boolean = a.tag.isProtected
+    def isProtectedThis: Boolean = a.tag.isProtectedThis
+    def isPublic: Boolean = a.tag.isPublic
+    def within: Sym = new Sym(a.symbol)
```

Comment:
How does one check if something is `private[foo]` in this API?

Reply:
I think `isPrivate && within == FooSymbol`? But haven't had any time to test the ergonomics of this new API

Reply:
I don't think that `isPrivate` covers this. There is a separate `PRIVATE_WITHIN` tag in `Accessibility`.